server/channels/store/sqlstore/sqlx_wrapper_test.go

```diff
@@ -90,3 +90,90 @@ func TestSqlX(t *testing.T) {
 		}
 	})
 }
+
+func TestSqlxSelect(t *testing.T) {
+	t.Run("Select Methods", func(t *testing.T) {
+		testDrivers := []string{
+			model.DatabaseDriverPostgres,
+			model.DatabaseDriverMysql,
+		}
+
+		for _, driver := range testDrivers {
+			settings, err := makeSqlSettings(driver)
+			if err != nil {
+				continue
+			}
+			*settings.QueryTimeout = 1
+			store := &SqlStore{
+				rrCounter:   0,
+				srCounter:   0,
+				settings:    settings,
+				logger:      mlog.CreateConsoleTestLogger(t),
+				quitMonitor: make(chan struct{}),
+				wgMonitor:   &sync.WaitGroup{},
+			}
+
+			require.NoError(t, store.initConnection())
+			defer store.Close()
+
+			t.Run("Select with "+driver, func(t *testing.T) {
+				var result []string
+				err := store.GetMaster().Select(&result, "SELECT 'test' AS col")
+				require.NoError(t, err)
+				require.Equal(t, []string{"test"}, result)
+			})
+
+			t.Run("SelectCtx with "+driver, func(t *testing.T) {
+				var result []string
+				err := store.GetMaster().SelectCtx(context.Background(), &result, "SELECT 'test' AS col")
+				require.NoError(t, err)
+				require.Equal(t, []string{"test"}, result)
+
+				// Test timeout
+				ctx, cancel := context.WithTimeout(context.Background(), 1)
+				defer cancel()
+				var query string
+				if driver == model.DatabaseDriverMysql {
+					query = "SELECT SLEEP(2)"
+				} else {
+					query = "SELECT pg_sleep(2)"
+				}
+				err = store.GetMaster().SelectCtx(ctx, &result, query)
```

Comment:
Is this test really necessary? You are effectively testing the database here.

Reply:
In this case we're testing the  context timeout, which does seem useful. If anything, I would probably remove the basic `"Select with "+driver` test above (and the similar one below) instead. :shrug: Up to you, I'll defer to your expertise on the usefulness. :)

Reply:
Oh yeah, indeed. Let's remove select with driver and selectBuilder with driver.